ecosystem/nft-metadata-crawler/src/asset_uploader/api/get_status.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    asset_uploader::api::GetStatusResponseSuccess,
+    models::asset_uploader_request_statuses_query::AssetUploaderRequestStatusesQuery, schema,
+};
+use ahash::AHashMap;
+use diesel::{
+    r2d2::{ConnectionManager, Pool, PooledConnection},
+    ExpressionMethods, PgConnection, QueryDsl, RunQueryDsl,
+};
+use tracing::debug;
+use uuid::Uuid;
+
+pub fn get_status(
+    pool: Pool<ConnectionManager<PgConnection>>,
+    request_id: &str,
+) -> anyhow::Result<AHashMap<String, GetStatusResponseSuccess>> {
+    let mut conn = pool.get()?;
+    let request_id = Uuid::parse_str(request_id)?;
+
+    let mut status_response = AHashMap::new();
+    let rows = query_status(&mut conn, &request_id)?;
+    for row in rows {
+        let status_code = row.status_code.map(|x| x as u16);
+
+        let response = if let Some(status_code) = status_code {
+            if status_code == 200 {
+                GetStatusResponseSuccess::Success {
+                    status_code: Some(status_code),
+                    cdn_image_uri: row.cdn_image_uri.unwrap_or_default(),
+                }
+            } else {
+                GetStatusResponseSuccess::Error {
+                    status_code: Some(status_code),
+                    error_message: row.error_message,
+                }
+            }
+        } else {
+            GetStatusResponseSuccess::Error {
+                status_code,
+                error_message: row.error_message,
+            }
+        };
```

Comment:
in the later implementation, if its a new request, both status code and error message is null, so here, it actually won't return any valuable. we might wanna change what we update to db when its a new request pending processing, like status code can be 202 accepted, to be more informative

Reply:
good idea, gonna just set default value to 202